### PR TITLE
feat(meshservice): Add meshservice checkbox to tabs component

### PR DESCRIPTION
**Pull request by johncowen**
Continuation of https://github.com/kumahq/kuma-website/pull/1940

Closes https://github.com/kumahq/kuma-website/issues/1919

rendered: https://deploy-preview-1985--kuma.netlify.app/docs/dev/policies/meshtimeout/#full-config-applied-to-inbound-and-outbound-of-specific-service

---

### CODE REVIEW

File: jekyll-kuma-plugins/lib/jekyll/kuma-plugins/liquid/tags/policyyaml.rb [hunk 1]
```diff
@@ -47,67 +65,115 @@ def process_hash(hash, remove_suffix, rename_suffix)
...
+<div class=\"meshservice\">
+  I am using <a href=\"\">MeshService</a> <input type=\"checkbox\">
+</div>
+#{kube_style1_content}
+#{kube_style2_content}
 {% endtab %}
 {% tab #{@tabs_name} Universal %}
-#{uni_content}
+<div class=\"meshservice\">
+  I am using <a href=\"\">MeshService</a> <input type=\"checkbox\">
```
Comment (johncowen) [inline jekyll-kuma-plugins/lib/jekyll/kuma-plugins/liquid/tags/policyyaml.rb:159]: We'll need to add the URL that goes in here still.

Comment (slonka) [reply jekyll-kuma-plugins/lib/jekyll/kuma-plugins/liquid/tags/policyyaml.rb]: Done

File: jekyll-kuma-plugins/lib/jekyll/kuma-plugins/liquid/tags/policyyaml.rb [hunk 2]
```diff
@@ -1,41 +1,59 @@
...
-                sp = item.split('=')
-                @params[sp[0]] = sp[1] unless sp[1] == ''
+              key, value = item.split('=')
+              @params[key] = value unless value == ''
+            end
+          end
+
+          # Function to transform targetRef based on MeshService name (if needed)
+          def transform_target_ref(hash)
+            if hash.dig("spec", "targetRef", "kind") == "MeshService"
```
Comment (michaelbeaumont) [inline jekyll-kuma-plugins/lib/jekyll/kuma-plugins/liquid/tags/policyyaml.rb:22]: Shouldn't this change `spec.to[].targetRef`?

**Review by michaelbeaumont — APPROVED** [review]
LGTM, I guess support for backendRefs would be next?

**Comment by slonka** [comment]
> LGTM, I guess support for backendRefs would be next?

Yes, but I hope someone will help me out on this one

**Comment by slonka** [comment]
We need a guard for the thing to only be displayed in 2.9.x onwards

**Comment by jijiechen** [comment]
This looks quite cool.
One small question: why the checkbox is not at the beginning of the label but at the ending?

**Review by michaelbeaumont — APPROVED** [review]

File: app/_src/policies/meshtimeout.md [hunk 3]
```diff
@@ -219,7 +219,7 @@ spec:
 This timeout configuration will be applied to all inbound connections to `frontend` and outbound connections
 from `frontend` to `backend` service
 
-{% policy_yaml example4 %}
+{% policy_yaml example4 use_meshservice=true %}
```
Comment (slonka) [inline app/_src/policies/meshtimeout.md:222]:
```suggestion
{% policy_yaml example4 %}
```

**Pull request merged**